src/pages/PostDeliveryDetailsScreen/PostDeliveryDetailsScreen.tsx

```diff
@@ -0,0 +1,102 @@
+import { forwardRef, useContext } from 'react';
+import { Form } from 'react-final-form';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Tooltip from '@mui/material/Tooltip';
+import { Select } from 'mui-rff';
+import CustomFormSpy from 'components/CustomFormSpy';
+import FormScreens from 'constants/formScreens';
+import WizardFormContext, {
+  InitialFormValuesType,
+} from 'contexts/WizardFormContext';
+import { validateIsRequired } from 'utils/validation';
+import PostCompanies from 'constants/postCompanies';
+import POST_COMPANIES_OPTIONS from 'constants/postCompaniesOptions';
+import POST_OFFICES_OPTIONS from 'constants/postOfficesOptions';
+
+const getPostOfficeOptions = (postCompany: PostCompanies) => {
+  switch (postCompany) {
+    case PostCompanies.UKRPOSHTA:
+      return POST_OFFICES_OPTIONS.ukrposhta;
+    case PostCompanies.NOVA_POST:
+      return POST_OFFICES_OPTIONS.novaPost;
+    case PostCompanies.MEEST_POSHTA:
+      return POST_OFFICES_OPTIONS.meestPoshta;
+    default:
+      throw new Error('New post company found');
+  }
+};
+
+interface Props {
+  postCompany?: PostCompanies | null;
+}
+
+const PostOfficeSelection = forwardRef(({ postCompany }: Props, ref) => (
```

Comment:
You don't need `forwardRef` and conditional rendering here. But in order to show a tooltip message, you need to wrap contents of tooltip in html element, e.g. `span`. Check this [example](https://mui.com/material-ui/react-tooltip/#disabled-elements)